index.js

```diff
@@ -197,6 +201,11 @@ class KeyringController extends EventEmitter {
   addNewKeyring(type, opts) {
     const Keyring = this.getKeyringClassForType(type);
     const keyring = new Keyring(opts);
+    if (type === KEYRINGS_TYPE_MAP.HD_KEYRING) {
```

Comment:
We don't want to do this if `opts` contains a `mnemonic`. This would erase the old SRP and generate a new one :grimacing:

Reply:
I'm not really concerned about coverage so much as that the change introduced in this PR is tested. This would not be caught by a coverage report - all of these lines were already covered.

Reply:
Yes sorry I mean **bump** & **improve** coverage!

Reply:
I can add a test for this particular issue in this PR though!

Reply:
done [here](https://github.com/MetaMask/KeyringController/pull/132/commits/8bfcad706a7f0f20905c145277f53916c522cc20)!